__tests__/commands/install/integration.js

```diff
@@ -437,6 +437,24 @@ test.concurrent('install with file: protocol as default', (): Promise<void> => {
   });
 });
 
+test.concurrent('install file: dedupe dependencies', async (): Promise<void> => {
+  await runInstall({}, 'install-file-dedupe-dependencies', async (config, reporter) => {
+    expect(await fs.exists(path.join(config.cwd, 'node_modules', 'a', 'node_modules'))).toEqual(false);
+
+    // Add b as a dependency of c, as an absolute path
+    const pkgJsonContents = await fs.readFile(path.join(config.cwd, 'c/package.json'));
+    const pkgJson = JSON.parse(pkgJsonContents);
+    pkgJson.dependencies = {b: `file:${path.join(config.cwd, 'b')}`};
+    await fs.writeFile(path.join(config.cwd, 'c/package.json'), JSON.stringify(pkgJson));
+
+    const lockfile = await Lockfile.fromDirectory(config.cwd);
+    const reInstall = new Install({}, config, reporter, lockfile);
+    await reInstall.init();
+
+    expect(await fs.exists(path.join(config.cwd, 'node_modules', 'c', 'node_modules'))).toEqual(false);
```

Comment:
Again, what are we testing? I'm guessing you are testing now `c` also doesn't have any sub-dependencies _even if_ we added `b` to it. I'd suggest creating a separate test case for this whole scenario. It would make your test cases more understandable and atomic so if something goes wrong in the future, we won't look at this test and try to decipher what exactly went wrong :)